### PR TITLE
Background music

### PR DESCRIPTION
**Pull request by AlfredoCuevas**
High Level Goals: 
Add sounds to the game in a way that allows programmers and non-programmers to interact with the AudioSource tools provided by Unity.

Important Files Added: 
**AudioManager.cs**: Manages a list of AudioSources provided before the game is executed. Provides public functions that allow control over the sounds given. 
**AudioTest.cs**: A simple script that calls functions from AudioManager.Instance to verify that they function as intended. 

Files Changed: 
**PlayerJump.cs**: I added one line of code that plays the jump sound when the jump force is applied to the player. 

New Features: 
The AudioManager Prefab

**AudioManager.cs**: 
* This script creates a static Instance that will persist throughout every scene. On Awake() it will look through all of its children and create a list of AudioSources. These AudioSources can be manipulated by calling public functions with the name of the AudioSource's GameObject. 
* Current Noteworthy Functions Include:
    -IsPlaying(): Returns a bool
    -Play() / Stop(): Plays or Stops given song
    -PlayOneShot(): Allows multiple objects to play the same sound. Sound cannot be stopped
    -FadeIn() / FadeOut(): Will play / stop a song while adjusting volume over a certain amount of time
* Only one AudioManager needs to be created once the Scenes are linked to each other. 

**AudioTest.cs**:
* This script will read input from the keyboard and call the public functions available through the AudioManager
* The key controls can be seen in the inspector when looking at the script
* A test scene (BackgroundMusicTest) was created to allow for easier testing. The default buttons should be J (play), K (Stop), I (FadeIn), O (FadeOut). The functions were all called on the "Gameplay Theme" which plays on Awake()

**Testing**: 
Go into the scene BackgroundMusicTest. Run Unity and check that the Play, Stop, Volume, and Fade In / Fade Out functions work as expected. Look at the (Audio Test Object) in the inspector to see which keys to press. (By default the keys should be J,K, Up Arrow, Down Arrow, I, O ). 

**Adding A Sound To the Game**: 
To add a sound you need to create an Empty GameObject as a child of the AudioManager. In the object add the component AudioSource. Then drag and drop the AudioClip from the files into the AudioSource. Make sure to give the GameObject an unique easily understandable name. 
Once this is done you should be able to call AudioManager functions by providing the GameObject's name.

### CODE REVIEW

File: Assets/Scripts/AudioTest.cs [hunk 1]
```diff
@@ -0,0 +1,45 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class AudioTest : MonoBehaviour
+{
+    [SerializeField]
+    private KeyCode play;
```
[the remark this answers is not in the document]
Comment (AlfredoCuevas) [reply Assets/Scripts/AudioTest.cs]: My plan was to delete the AudioTest after testing so I didn't put as much attention to the names. But I can go in and change it.

File: Assets/Scenes.meta [hunk 2]
```diff
@@ -1,5 +1,5 @@
 fileFormatVersion: 2
-guid: 4f704ae4b4f98ae41a0bce26658850c1
+guid: 4cf5f5d5e8700f644a2fd56a1f9baf92
```
[the remark this answers is not in the document]
Comment (AlfredoCuevas) [reply Assets/Scenes.meta]: I talked to him and he said that when you press Play Unity seems to fix any conflicts with meta files.

**Comment by JamesCoulter** [comment]
Odd behavior when "fade out" is pressed before fading in is done, and vice versa. Seems to cause the volume to change at a slower rate, and the "Is music playing" bool no longer outputs correctly. Otherwise, it all looks good.

**Comment by JamesCoulter** [comment]
Changes look like they work, I'd say it's ready to merge in

**Pull request merged**